src/brevitas/core/function_wrapper/clamp.py

```diff
@@ -73,3 +78,143 @@ def __init__(self, min_val: float) -> None:
     @brevitas.jit.script_method
     def forward(self, x: Tensor):
         return x.clamp_min(self.min_val)
+
+
+class FloatClamp(brevitas.jit.ScriptModule):
+    """"
+    ScriptModule for clamping minifloat formats to their inf/NaN implementations.
+
+    Currently, inf/NaN codes have to be encoded through the mantissa.
+    I.e. setting inf to 1101.111 (E4M3) is not a valid code.
+    """
+
+    __constants__ = ['nan_values', 'inf_values', 'saturating']
+
+    def __init__(
+            self,
+            exponent_bit_width: Tensor,
+            mantissa_bit_width: Tensor,
+            exponent_bias: Tensor,
+            nan_values: Optional[Tuple[str]] = None,
+            inf_values: Optional[Tuple[str]] = None,
+            saturating: bool = False) -> None:
+        super(FloatClamp, self).__init__()
+
+        self.exponent_bit_width = torch.tensor(exponent_bit_width)
```

Comment:
If the input is a Tensor, casting to a tensor is not needed.
Also, no need to save it in self since we just need to pass these values around